gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -463,7 +461,7 @@ private void deleteFromExecutor(Spec spec, Properties headers) {
    Deletes spec from flowCatalog if it is an adhoc flow (not containing a job schedule)
  */
   private void deleteSpecFromCatalogIfAdhoc(FlowSpec flowSpec) {
-    if (!flowSpec.getConfig().hasPath(ConfigurationKeys.JOB_SCHEDULE_KEY)) {
+    if (!flowSpec.isScheduled()) {
```

Comment:
big improvement! :)

did we have this method all along, but just weren't using it here?

Reply:
Yes, I went to add this util method thinking it would be useful and found it already exists